### ISSUE TITLE
VS Code June 2018; fail to compile

### ISSUE THREAD

**Issue by Zipgit** [Microsoft/vscode#53986]
Despite a lot of effort, my updated June 2018 version refuses to compile my code. I've let it run the Build task for up to 15 minutes (!!), it chugs along, but no file is returned. 
I am getting no build-time diagnostic errors, the PATH is correct (in VSC as well as the Environment Variables System PATH). I am using MinGW, & have installed the C++ and Clang extensions. The platform is Win7 64 bit, fully up-to-date.

**Comment by dbaeumer** [comment]
@Zipgit is there anything special in the terminal that runs the build task. When you do Tasks > Run Task is the build task to executed listed there?

If yes, can you provide me with a GitHub repository I can clone that demos the behavior?

**Comment by Zipgit** [comment]
Dirk:
Thanks for a fast reply. I've created a repository
https://github.com/Zipgit/FreshWorld
As you can see I am new to C++ so was using a "hello world" to make sure the installation was running properly. As noted, it WAS working on my platform, now it is failing to compile. Strange.

**Comment by dbaeumer** [comment]
@Zipgit we broke something around tasks that might have caused this. We fix it and it is available in the latest insider. Would you mind testing this with the latest insider build before I dig into it?

**Comment by Zipgit** [comment]
I did as you asked. 

Here is the summary of results:

- Progress was further; instead of hanging indefinitely it finished the compilation and gave the "terminal will be reused....please press any key...."
**- **No *.exe file was generated.****

I also got a message after installing the "insiders" version, indicating that I should consider _"switching to the Users Setup Distribution of Code -- Insiders for Windows"_
![recursive message](https://user-images.githubusercontent.com/36453740/42595762-1ab2d478-8508-11e8-80ca-721728840189.JPG)

This just sends the user around in a recursive loop. Perhaps there should be some code to make sure the pop up only appears when a non-Insiders version is running?

**Comment by dbaeumer** [comment]
@Zipgit that is good news. Can you let me know that task you are executing now. If this comes from the CPP extension then you need to talk to them if you have further issues.

@joaomoreno please see the comment above about 'switching to the Users Setup Distribution of Code -- Insiders for Windows'

**Comment by joaomoreno** [comment]
The popup is correct. Click the gear and `Do not show again` if you don't want to see it again.

**Comment by Zipgit** [comment]
Dirk: _"Can you let me know that task you are executing now. If this comes from the CPP extension then you need to talk to them if you have further issues."_ The task I am executing is the same one that is in the Repository I made. Yes, I am of course using the VSCode CPP extension....but  who are the "them" that I need to speak to???
Joaomoreno--my only point about the popup was, that if you are already running the "insiders" version, Why would it tell you to switch to it? Surely there could be some internal check to see if the version being run was the latest (stable) build, or a nightly release.

**Comment by dbaeumer** [comment]
@Zipgit actually the task works for me as expected. It generates `a.out` file which is the standard for g++. Please consult the g++ command line params on how to change this.

I am closing the issue. Please ping if you think there is still something not working.

**Comment by Zipgit** [comment]
Hello, Dirk:

 

Yes, I think there is something still not working. 

 

I uninstalled and reinstalled the MinGW c++ compiler and it still does not work.

I have followed all the instructions for setting up the PATHs etc as well as the commands in the Launch.json and Task.json====the ones which worked previously==== and it still will not compile for me.

 

Or, perhaps, it is compiling but not writing the Executable to the correct folder ( am using "${workspaceFolder}\a.exe")

 

If you spare a thought on this, it would be appreciated. It's difficult getting started down a road when you have no legs.

**Comment by dbaeumer** [comment]
What happens if you execute the command directly in a terminal ?

**Comment by Zipgit** [comment]
Hello, Dirk:

 

Thanks for helping me with this problem

This is a new experience for me as I am new to C++ but I think I've done what you've asked properly:

 

·        I open a Windows DOS CMD window; 

·        I navigated to the directory where the "FreshWorld.cpp" code resides

·        I used the command...

·        g++ -o FreshWorld.exe FreshWorld.cpp

·        As i understand the command, this command line should compile FreshWorld.cpp, generating FreshWorld.exe, and it should drop that file into the same folder.

 

When I executed the command, it seemed to run without any problems (no error messages were generated) BUT....there is no *.exe file generated in that folder. If the *.exe file was stored somewhere else....then I don't even know where to look.

 

Michael

**Comment by dbaeumer** [comment]
@Zipgit if this happens in a standalone terminal as well there is little VS Code can do to help here. All VS Code does is ti launch these commands in a terminal. Have you tried to ask the question on StackOverflow?

**Comment by Zipgit** [comment]
Hello, Dirk:

Do you have any thoughts about my problem? This is a very strange (and irritating!) problem as I had followed a set of setup instructions previously (with the previous stable release of Visual Studio Code), and everything worked fine. Now and despite having uninstalled VSCode, and having reinstalled the C++ components of MinGW, I cannot get VSCode to move forward in the precisely predictable way that the old VSCode did...

 

Michael G

**Comment by dbaeumer** [comment]
@Zipgit may be I misunderstand you but execute `g++ -o FreshWorld.exe FreshWorld.cpp` in a standalone terminal is not producing a `FreshWorld.exe` for you. Right?. If this is the case I can't see how having VS Code installed can cause this behaviour.

**Comment by Zipgit** [comment]
Dirk:

 

Yes, I am running from the file-containing directory as such:

      c:\CPlusPlus\g++ -o FreshWorld.exe FreshWorld.cpp

from the Windows command line.

 

When I hit [RETURN], it takes about 10 seconds, and returns 

      C:\CplusPlus>

 

The executable cannot be found in the "workspace" directory, C:\CPlusPlus

 

I have executed a Windows search for FreshWorld.exe within all of my USER folders, but it does not exist there.

 

I find it curious / surprising that there is no error message, as g++ should be generating the *.exe file.....and trying to store it, somewhere!

 

@Zipgit

**Comment by dbaeumer** [comment]
I can understand your frustration around this but I am not sure what you want me to do. This all happens outside of VS Code. Have you tried to post a question on Stack Overflow. There is a mingw tag.

I am open to investigate the issue if this happens only when executing the task in VS Code itself.

**Comment by Zipgit** [comment]
Dirk:

 

No, I don't have any expectations of you "doing" anything....I was just soliciting suggestions. I did in fact post this to Stack Overflow; originally as a question including VS Code but yesterday I posted another question related specifically to C++ compilation with MinGW.

 

So far (and it has been a week or more) there have been no answers, nor any suggestions.

 

Thanks again for your help to date. If I discover that there's an issue definitely connected to VSCode I will get in touch with you again.

 

Michael G
